### PR TITLE
Updated Commerce branch to G&K

### PR DESCRIPTION
**Pull request by xlenstra**
Fourth part of the split of #4084 into multiple PR's for each branch.
This PR updates the Commerce branch to have G&K effects instead of those currently present.
It also modifies the existing policy uniques so that they are more moddable.

Added uniques:
- "+[int] happiness from each type of luxury resource"

Deprecated uniques:
- "+1 happiness from each type of luxury resource"
- "All military naval units receive +1 movement and +1 sight"

After merging this PR some problems might arise when updating an existing game. This is, as the commerce tree contains multiply differently named policies, and shuffles those with the same name around quite a bit. I have fixed this by automatically converting removed policies to the new policies, but this might result in some policy trees looking weird. It is, for example, possible to only have both the second tier policies of the branch, without unlocking the first tier.

### CODE REVIEW

**Comment by xlenstra** [comment]
I'll fix the merge conflicts once master builds again

**Comment by yairm210** [comment]
Master builds

**Comment by xlenstra** [comment]
Fixed merge conflicts

File: android/assets/jsons/Civ V - Vanilla/Buildings.json [hunk 1]
```diff
@@ -203,7 +203,7 @@
 		"greatPersonPoints": {"gold": 1},
 		"isWonder": true,
 		"providesFreeBuilding":  "Lighthouse",
-		"uniques": ["Must be next to [Coast]", "All military naval units receive +1 movement and +1 sight"],
+		"uniques": ["Must be next to [Coast]", "+[1] movement and +[1] sight for all [military water] units"],
```
Comment (yairm210) [inline android/assets/jsons/Civ V - Vanilla/Buildings.json:206]: Split movement and sight into 2 separate uniques

Comment (yairm210) [reply android/assets/jsons/Civ V - Vanilla/Buildings.json:206]: I think that they should be capitalized as well, I can't recall what we usually use

File: core/src/com/unciv/logic/map/MapUnit.kt [hunk 2]
```diff
@@ -141,14 +141,25 @@ class MapUnit {
...
+        // Deprecated since 3.14.17
+            if (type.isMilitary() && type.isWaterUnit() && civInfo.hasUnique("All military naval units receive +1 movement and +1 sight")) {
+                movement += 1
+            }
+        //
+        
         for (unique in civInfo.getMatchingUniques("+[] Movement for all [] units"))
             if (matchesFilter(unique.params[1]))
                 movement += unique.params[0].toInt()
+        for (unique in civInfo.getMatchingUniques("+[] movement for every []")) {
```
Comment (yairm210) [inline core/src/com/unciv/logic/map/MapUnit.kt:159]: "+[] Movement for all [] units" - that way we can have it match unitFiltet

**Pull request merged**